matchzoo/layers/spatial_gru.py

```diff
@@ -0,0 +1,254 @@
+"""An implementation of Spatial GRU Layer."""
+import typing
+
+from keras import backend as K
+from keras.engine import Layer
+from keras.layers import Permute
+from keras.layers import Reshape
+from keras.layers import activations
+from keras.layers import initializers
+
+from tensorflow.python.ops import control_flow_ops
+from tensorflow.python.ops import tensor_array_ops
+
+
+class SpatialGRU(Layer):
+    """
+    Spatial GRU layer.
+
+    :param units: Number of SpatialGRU units.
+    :param normalize: Whether to L2-normalize samples along the
+        dot product axis before taking the dot product.
+        If set to True, then the output of the dot product
+        is the cosine proximity between the two samples.
+    :param init_diag: Whether to initialize the diagonal elements
+        of the matrix.
+    :param activation: Activation function to use. Default:
+        hyperbolic tangent (`tanh`). If you pass `None`, no
+        activation is applied (ie. "linear" activation: `a(x) = x`).
+    :param recurrent_activation: Activation function to use for
+        the recurrent step. Default: hard sigmoid (`hard_sigmoid`).
+        If you pass `None`, no activation is applied (ie. "linear"
+        activation: `a(x) = x`).
+    :param use_bias: Boolean, whether the layer uses a bias vector.
+    :param kernel_initializer: Initializer for the `kernel` weights
+        matrix, used for the linear transformation of the inputs.
+    :param recurrent_initializer: Initializer for the `recurrent_kernel`
+        weights matrix, used for the linear transformation of the
+        recurrent state.
+    :param bias_initializer: Initializer for the bias vector.
+    :param kwargs: Standard layer keyword arguments.
+
+    Examples:
+        >>> import matchzoo as mz
+        >>> layer = mz.layers.SpatialGRU(units=50,
+        ...                              normalize=True)
+        >>> num_batch, channel, left_len, right_len = 5, 10, 3, 2
+        >>> layer.build([num_batch, channel, left_len, right_len])
+
+    """
+
+    def __init__(self,
+                 units: int = 50,
+                 normalize: bool = False,
+                 init_diag: bool = False,
+                 activation: str = 'tanh',
+                 recurrent_activation: str = 'hard_sigmoid',
```

Comment:
Can we add another hyper-parameter to indicate the direction of the spatialGRU? for example, `lt` (i.e., left top) indicates the scanning from left top to right bottom, and `rb` (i.e., right bottom) indicates the scanning from right bottom to left top.

Reply:
Hyper-parameter `direction` has been added to MatchSRNN model.